mitxgraders/helpers/validatorfuncs.py

```diff
@@ -96,20 +97,19 @@ def all_unique(iterable):
     True
 
     Raises an error if any items are duplicated:
-    >>> iterable = ['a', 0, '1', 'a', '5', 0, '0']
+    >>> iterable = ['a', 0, '1', 'a', '5', 0, 0, '0']
     >>> try:
     ...     all_unique(iterable)
     ... except Invalid as error:
     ...     print(error)
     items should be unique, but have unexpected duplicates: ['a', 0]
     """
-    seen = set()
-    duplicates = set()
+    counts = defaultdict(int)
```

Comment:
The `set` version of `duplicates` wasn't good enough, because it is unordered and generally cannot be sorted by `sorted` (e.g., in the doctest above where strings and integers are incomparable for sorting).

So this version keeps `duplicates` sorted by order in the original iterable object.